### PR TITLE
allow wildcard overlap with literal status code

### PR DESCRIPTION
**Pull request by sakari**
see https://swagger.io/specification/

fixes https://github.com/smartlyio/oats/issues/211

### CODE REVIEW

File: src/status-codes.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    seen.add(code);
+  });
+  const record = new Map();
+  const matchers: RegExp[] = codes.sort().map(c => {
+    if (c === 'default') {
+      return /.*/;
+    }
+    return new RegExp(c.replace(/XX$/, '..'));
+  });
+  for (const code of statusCodes) {
```
Comment (ganymedes01) [inline src/status-codes.ts:20]: how would this behave if we have two matches? something like:
```yml
default:
  description: rest
404:
  description: not found
4XX:
  description: some error
```

all the `4XX` codes would also match `default` and might end up in either depending on the order in which they were declared in the file I guess 😣

Comment (sakari) [reply src/status-codes.ts:20]: https://github.com/smartlyio/oats/pull/213/files#diff-4919a218fd041747e2f69f1ec399a576035ec3018691faa95eefcc02a5f264c0R14 sorts the status codes to first have numbers and then letters so 400 gets sorted before 4XX which get sorted before default.

thats also the reason for the old school loop here to allow breaking from the loop when we get the first match

Comment (ganymedes01) [reply src/status-codes.ts:20]: oh nice! didn't notice the `sort()` there 👍

**Review by ganymedes01 — APPROVED** [review]
looks good 👌

**Comment by sakari** [comment]
I'll add a better test to catch pin point the magic with sort still

**Pull request merged**